src/driver.rs

```diff
@@ -108,6 +110,131 @@ impl rustc_driver::Callbacks for ClippyCallbacks {
     }
 }
 
+fn describe_lints() {
+    use lintlist::*;
+    use std::collections::HashSet;
+
+    println!(
+        "
+Available lint options:
+    -W <foo>           Warn about <foo>
+    -A <foo>           Allow <foo>
+    -D <foo>           Deny <foo>
+    -F <foo>           Forbid <foo> (deny <foo> and all attempts to override)
+
+"
+    );
+
+    let lint_level = |lint: &Lint| {
+        LINT_LEVELS
+            .iter()
+            .find(|level_mapping| level_mapping.0 == lint.group)
+            .map(|(_, level)| level)
+            .unwrap()
+    };
+
+    let mut lints: Vec<_> = ALL_LINTS.iter().collect();
+    // The sort doesn't case-fold but it's doubtful we care.
+    lints.sort_by_cached_key(|x: &&Lint| (lint_level(x), x.name));
+
+    let max_name_len = lints
+        .iter()
+        .map(|lint| lint.name.len())
+        .map(|len| len + "clippy::".len())
+        .max()
+        .unwrap_or(0);
+
+    let padded = |x: &str| {
+        let mut s = " ".repeat(max_name_len - x.chars().count());
+        s.push_str(x);
+        s
+    };
+
+    let scoped = |x: &str| format!("clippy::{}", x);
+
+    let lint_groups: HashSet<_> = lints.iter().map(|lint| lint.group).collect();
+
+    println!("Lint checks provided by clippy:\n");
+    println!("    {}  {:7.7}  meaning", padded("name"), "default");
+    println!("    {}  {:7.7}  -------", padded("----"), "-------");
+
+    let print_lints = |lints: &[&Lint]| {
+        for lint in lints {
+            let name = lint.name.replace("_", "-");
+            println!("    {}  {:7.7}  {}", padded(&scoped(&name)), lint_level(lint), lint.desc);
+        }
+        println!("\n");
+    };
+
+    print_lints(&lints);
+
+    let max_name_len = std::cmp::max(
+        "warnings".len(),
```

Comment:
is there a `clippy::warnings` group?

Reply:
No, there's only `clippy::all` which is all warn-by-default and deny-by-default lints. You can find all available lint groups in the README:

> * clippy::all
> * clippy::correctness (deny-by-default)
> * clippy::style (warn)
> * clippy::complexity (warn)
> * clippy::perf (warn)
> * clippy::pedantic (allow)
> * clippy::nursery (allow)
> * clippy::cargo (allow)

There's also `clippy::restriction`, which is allow-by-default and lints where it doesn't make sense to enable the whole group.

Reply:
okay, I bring this up because rustc has a hard coded bit for their warnings group because its otherwise not included with their other lint groups, it looks like we've got everything in our groups except `all`, so I'm going to treat all like our version of warning and include it in the output the same way that rustc does.